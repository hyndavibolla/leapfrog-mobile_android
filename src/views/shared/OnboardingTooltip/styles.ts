import { css } from '@emotion/native';

export const styles = {
  title: css`
    margin-bottom: 8px;
  `,
  buttonContainer: css`
    position: absolute;
    right: -12px;
    top: -16px;
  `
};
