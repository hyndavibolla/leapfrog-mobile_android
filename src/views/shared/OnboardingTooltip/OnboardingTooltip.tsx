import React from 'react';
import { View } from 'react-native';

import { Title, TitleType } from '_components/Title';
import { Text } from '_components/Text';

import { TooltipButton } from '_components/TooltipButton';
import { useTestingHelper } from '_utils/useTestingHelper';

import { styles } from './styles';

export interface Props {
  onPress: () => void;
  isFlashy: boolean;
  title: string;
  subtitle?: string;
}

export const OnboardingTooltip = ({ title, subtitle, onPress, isFlashy }: Props) => {
  const { getTestIdProps } = useTestingHelper('onboarding-tooltip');

  return (
    <View {...getTestIdProps('container')}>
      <Title type={TitleType.HEADER} style={styles.title}>
        {title}
      </Title>
      <View style={styles.buttonContainer}>
        <TooltipButton flashy={isFlashy} altInactive onPress={onPress} />
      </View>
      {subtitle && <Text {...getTestIdProps('subtitle')}>{subtitle}</Text>}
    </View>
  );
};
