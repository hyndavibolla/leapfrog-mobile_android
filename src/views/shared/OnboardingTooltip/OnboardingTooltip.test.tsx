import React from 'react';
import { render } from '@testing-library/react-native';

import { OnboardingTooltip, Props } from './OnboardingTooltip';

describe('OnboardingTooltip', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      title: 'Title',
      onPress: jest.fn(),
      isFlashy: false
    };
  });

  it('should render with title flashy', () => {
    const { toJSON } = render(<OnboardingTooltip {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without subtitle', () => {
    const { queryByTestId } = render(<OnboardingTooltip {...props} />);
    expect(queryByTestId('onboarding-tooltip-subtitle')).toBeFalsy();
  });

  it('should render with subtitle', () => {
    props.subtitle = 'Subtitle';

    const { getByTestId } = render(<OnboardingTooltip {...props} />);
    expect(getByTestId('onboarding-tooltip-subtitle')).toBeTruthy();
  });
});
