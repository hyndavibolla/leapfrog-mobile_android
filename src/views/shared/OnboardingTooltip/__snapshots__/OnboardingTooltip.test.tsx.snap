// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingTooltip should render with title flashy 1`] = `
<View
  accessibilityLabel="onboarding-tooltip-container"
  accessible={true}
  testID="onboarding-tooltip-container"
>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 20,
          "lineHeight": 30,
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
    type="titleHeader"
  >
    Title
  </Text>
  <View
    style={
      Object {
        "position": "absolute",
        "right": -12,
        "top": -16,
      }
    }
  >
    <View
      accessibilityLabel="tooltip-button-container"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 54,
          "justifyContent": "center",
          "opacity": 1,
          "position": "relative",
          "width": 54,
        }
      }
      testID="tooltip-button-container"
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#E2E3E3",
            "borderBottomLeftRadius": 50,
            "borderBottomRightRadius": 50,
            "borderTopLeftRadius": 50,
            "borderTopRightRadius": 50,
            "justifyContent": "center",
            "marginBottom": 4,
            "marginLeft": 4,
            "marginRight": 4,
            "marginTop": 4,
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
            "width": 30,
          }
        }
      >
        TooltipInactive
      </View>
    </View>
  </View>
</View>
`;
